js/utils.js

```diff
@@ -1511,6 +1511,51 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
     });
   }
 
+  function setFilterValues(data) {
+    return new Promise(function(resolve) {
+      if (!data.filterOptions.length) {
+        return resolve();
+      }
+
+      data.filterOptions.forEach(function(item) {
+        switch (item.valueType) {
+          case 'user-profile-data':
+            Fliplet.User.getCachedSession()
+              .then(function(options) {
+                var entries = options.entries;
+
+                if (options && options.entries) {
+                  if (entries.dataSource && entries.dataSource.data.hasOwnProperty(item.fieldValue)) {
+                    item.value = entries.dataSource.data[item.fieldValue];
+                    return resolve();
```

Comment:
If there are multiple filter options, this one single `resolve()` on the first loop will resolve the `setFilterValues()` promise. This won't work.